src/main/java/net/sf/jabref/gui/entryeditor/EntryEditorTabRelatedArticles.java

```diff
@@ -0,0 +1,165 @@
+
+    private static final String RECOMMENDATION_SEPERATOR = "<br>";
+
+    private static final Log LOGGER = LogFactory.getLog(EntryEditorTabRelatedArticles.class);
+
+
+    /**
+     * Takes the selected entry, runs a request to Mr. DLib and returns the recommendations as a JEditorPane
+     * @param selectedEntry
+     */
+    public EntryEditorTabRelatedArticles(BibEntry selectedEntry) {
+        this.selectedEntry = selectedEntry;
+        this.setContentType("text/html");
+        this.setEditable(false);
+        registerHyperlinkListener();
+        setDefaultContent();
+    }
+
+
+    /**
+     * Takes a List of html snippets and sets it in the JEditorPane
+     * @param list
+     */
+    public void setHtmlText(List<String> list) {
+        StringBuffer htmlContent = new StringBuffer();
+        htmlContent.append("<html><head><title></title></head><body bgcolor='#ffffff'><font size=8>");
+        for (String snippet : list) {
+            if (snippet != null) {
+                htmlContent.append(snippet);
+                htmlContent.append(RECOMMENDATION_SEPERATOR);
+            }
+        }
+        htmlContent.append("</font></body></html>");
+        this.setText(htmlContent.toString());
+    }
+
+    /**
+     * While the fetcher gets the Recommendations from MR. DLib this content indicates
+     * that the search is in Progress
+     */
+    private void setDefaultContent() {
+        StringBuffer htmlContent = new StringBuffer();
+
+        //What is the best way to include that gif?
+        URL url = IconTheme.getIconUrl("mdlloading");
+        htmlContent.append("<html><head><title></title></head><body bgcolor='#ffffff'><font size=5>");
+        htmlContent.append("Loading Recommendations for ");
+        htmlContent.append(selectedEntry.getLatexFreeField("title").get());
+        htmlContent.append("<br><img width=\"100\" height=\"100\" src=\"" + url + "\"></img>");
+        htmlContent.append("</font></body></html>");
+        this.setText(htmlContent.toString());
+    }
+
+    /**
+     * Makes the Hyperlinks clickable. Opens the link destination in a Browsertab
+     */
+    private void registerHyperlinkListener() {
+        this.addHyperlinkListener(new HyperlinkListener() {
+
+            @Override
+            public void hyperlinkUpdate(HyperlinkEvent e) {
+                if (e.getEventType() == HyperlinkEvent.EventType.ACTIVATED) {
+                    try {
+                        if (e.getURL() != null) {
+                            new JabRefDesktop().openBrowser(e.getURL().toString());
+                        }
+                    } catch (IOException e1) {
+                        LOGGER.error(e1.getMessage(), e1);
+                    }
+                }
+            }
+        });
+    }
+
+
+    /**
+     * Starts a Fetcher getting the recommendations form Mr. DLib
+     * @param selectedEntry
+     */
+    public void requestRecommendations() {
+        try {
```

Comment:
The special thing about Mr. DLib is that we have manifold algorithms that follow different approaches to calculate the recommendations. So requesting the recommendations once, does not mean that the second time you get the same recommendations. Especially because we constantly extend our database with new documents. A caching would not allow to deliver up-to-date recommendations.